# Практическое задание 5

## по теме: "Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование"

**Задача 1.**

Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

$$T_{1,2} = \overline{X} \pm \frac{s_0}{\sqrt{n}} \cdot c_{\gamma}$$

где $T_1$, $T_2$ — нижняя и верхняя границы доверительного интервала,

$\overline{X}$  — выборочное среднее арифметическое, $s_0$ — среднее квадратичное отклонение по выборке (несмещенное),

$n$ — размер выборки,

$\gamma$ — доверительная вероятность.

$c_{\gamma} = Ф^{-1} \frac{(1 + \gamma)}{2}$ — обратное значение функции стандартного нормального распределения.

Надежность = 0,95

Альфа = 5%

Вероятность 1-0,05/2 = 0,975

z(альфа/2) = 1,96

$$T_{1,2} = 80 \pm \frac{16}{\sqrt{256}} \cdot 1.96 = 80 \pm 1.96$$

$$ [78.04;   81.96]$$

**Задача 2.**

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

In [1]:
import pandas as pd

In [2]:
x = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]

In [3]:
x_df =  pd.DataFrame(x)

In [4]:
mean_x = x_df.sum() / x_df.count()
mean_x

0    6.59
dtype: float64

Считаем **среднее квадратичное отклонение** (несмещенное):

$$\sigma = \sqrt{\frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n-1}}$$

In [5]:
import numpy as np

In [6]:
x_std = np.sqrt(((x_df - mean_x)**2).sum() / (x_df.count() - 1))
x_std

0    0.450802
dtype: float64

$$T_{1,2} = \overline{X} \pm \frac{s_0}{\sqrt{n}} \cdot c_{\gamma} = 6.59 \pm \frac{0.45}{\sqrt{10}} \cdot 2.262 = 6.59 \pm 0.32$$

$$ [6.27;   6.91]$$

**Задача 3.**

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

Н0: шарики для подшипников имеют средний диаметр 17 мм

Н1: шарики для подшипников имеют средний диаметр 17.5 мм

z(табл) = 1.645

Стандартное отклонение равно 2/10 = 0.2

17.5 - 17.0 = 0.5

0.5 / 0.2 = 2.5

1.65 < 2.5

Следовательно, принимаем гипотезу H1 на уровне значимости 5%.

**Задача 4.**

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек.
Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно,
что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

H0: средний вес пачки печенья составляет 200 г.

t(табл) = 3.25 для доверительной вероятсти 0.5%

In [7]:
y = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]

In [8]:
df_y = pd.DataFrame(y)

In [9]:
df_y

0
0  202
1  203
2  199
3  197
4  195
5  201
6  200
7  204
8  194
9  190

In [10]:
mean_y = df_y.sum() / df_y.count()
mean_y

0    198.5
dtype: float64

In [11]:
y_std = np.sqrt(((df_y - mean_y)**2).sum() / (df_y.count() - 1))
y_std

0    4.453463
dtype: float64

Стандартное отклонение равно 4.45 / 3.162 = 1.41

200 - 198,5 = 1,5

1,5 / 1,41 = 1,06

1.06 < 3.25

Следовательно, принимаем нулевую гипотезу на уровне значимости 1%.